library/agent/InspectionStatistics.test.ts

```diff
@@ -462,6 +537,54 @@ t.test("it keeps track of aborted requests", async () => {
         total: 0,
         blocked: 0,
       },
+      blocked: {
+        total: 0,
+        userAgentList: {},
+        ipBlocklist: {},
+      },
     },
   });
+
+  clock.uninstall();
+});
+
+t.test("it keeps track of blocked requests", async () => {
+  const clock = FakeTimers.install();
+
+  const stats = new InspectionStatistics({
+    maxPerfSamplesInMemory: 50,
+    maxCompressedStatsInMemory: 5,
+  });
+
+  stats.onBlockedRequest({
+    match: "ipBlocklist",
+    key: "known_threat_actors/public_scanners",
+  });
+  stats.onBlockedRequest({ match: "userAgentList", key: "ai_data_scrapers" });
+
+  t.same(stats.getStats(), {
+    sinks: {},
+    startedAt: 0,
+    requests: {
+      total: 0,
```

Comment:
Blocked requests are not counted as total requests?

Reply:
```js
agent.getInspectionStatistics().onRequest();
```

is called in `createOnFinishRequestHandler`, that one will increment the total requests

Maybe we can refactor this later so that we have a single method to increment the counts together?